### PR TITLE
feat(adf-bootstrap): (#472) modify trust relations for roles :zap:

### PR DESCRIPTION
**Pull request by AndreasAugustin**
*Issue #472 *

*Description of changes:*
Modify the trust relations for the roles.
Now a recreation of the related roles is possible without the need of a recreation for the adf stacks in all accounts.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/global.yml [hunk 1]
```diff
@@ -34,8 +34,11 @@ Resources:
         Version: "2012-10-17"
         Statement:
           - Effect: Allow
+            Condition:
+              StringEquals:
```
Comment (StewartW) [inline src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/global.yml:38]:
```suggestion
              ArnEquals:
```

File: src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/global.yml [hunk 2]
```diff
@@ -201,21 +204,27 @@ Resources:
               - sts:AssumeRole
           - Effect: Allow
             Sid: "AssumeRoleLambda"
+            Condition:
+              StringEquals:
```
Comment (StewartW) [inline src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/global.yml:208]:
```suggestion
              ArnEquals:
```

File: src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/global.yml [hunk 3]
```diff
@@ -201,21 +204,27 @@ Resources:
...
             Principal:
               AWS:
-                - !Sub arn:${AWS::Partition}:iam::${DeploymentAccountId}:role/adf-codepipeline-role
+                - !Sub arn:${AWS::Partition}:iam::${DeploymentAccountId}:root
             Action:
               - sts:AssumeRole
             Condition:
-              ArnEquals:
-                "aws:SourceArn": !Sub "arn:${AWS::Partition}:codepipeline:${AWS::Region}:${DeploymentAccountId}:*"
+              Fn::And:
```
Comment (StewartW) [inline src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/global.yml:223]: This doesn't look right here. 
Are you trying to add both conditions in an AND fashion?

File: src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/global.yml [hunk 4]
```diff
@@ -231,9 +240,12 @@ Resources:
         Statement:
           - Effect: Allow
             Sid: "AssumeRole"
+            Condition:
+              StringEquals:
```
Comment (StewartW) [inline src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/global.yml:244]:
```suggestion
              ArnEquals:
```

File: src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/global.yml [hunk 5]
```diff
@@ -333,9 +345,12 @@ Resources:
...
```
Comment (StewartW) [inline src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/global.yml:349]: ArnEquals

**Review by StewartW — CHANGES_REQUESTED** [review]
I've requested a few changes and a comment. 

Thanks for raising the PR!

**Comment by StewartW** [comment]
Hey @AndreasAugustin thanks for raising this PR. It's a great solution. 

Regarding the multi-block condition, is there any reason you wouldn't do it the same way as defined here: https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_multi-value-conditions.html 

Rather than using the intrinsic AND operator.

**Comment by AndreasAugustin** [comment]
> Hey @AndreasAugustin thanks for raising this PR. It's a great solution.
> 
> Regarding the multi-block condition, is there any reason you wouldn't do it the same way as defined here: https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_multi-value-conditions.html
> 
> Rather than using the intrinsic AND operator.

Hi @StewartW thanks a lot for the review.
- fixed: `StringEquals` -> `ArnEquals`
- also moved from single quotes in some statements to double quotes
- `Fn::And` -> migrated to object based. I did not test it. At least `cfn-lint` is not complaining :+1:  I try to test it ASAP

**Review by StewartW — APPROVED** [review]

**Review by javydekoning — DISMISSED** [review]

File: src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/global.yml [hunk 6]
```diff
@@ -199,16 +202,27 @@ Resources:
...
+          - Effect: Allow
+            Sid: "AssumeRoleLambda"
+            Principal:
+              AWS:
+                - !Sub arn:${AWS::Partition}:iam::${DeploymentAccountId}:root
+            Action:
+              - sts:AssumeRole
+            Condition:
+              ArnEquals:
+                "aws:PrincipalArn": !Sub "arn:${AWS::Partition}:iam::${DeploymentAccountId}:role/adf-lambda-role"
```
Comment (sbkok) [inline src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/global.yml:214]: My mistake while merging, this is no longer in use. It got replaced by the the adf-automation roles instead.
```suggestion
```

**Review by sbkok — DISMISSED** [review]
Great fix, thanks for contributing this!

**Review by sbkok — APPROVED** [review]

**Review by javydekoning — APPROVED** [review]

**Pull request merged**